src/request/import/ImportWords.js

```diff
@@ -98,111 +126,106 @@ class ImportWords {
                     field._onBlur();
                 }, index * 50);
             }
-            recoveryWords.$fields.forEach((field, index) => {
+            this._recoveryWords.$fields.forEach((field, index) => {
                 putWord(field, testPassphrase[index], index);
             });
         };
     }
 
     run() {
-        this._entropy = null;
-        window.location.hash = ImportWords.Pages.PRIVACY_AGENT;
+        this._recoveryWords.setWords(new Array(24));
+        window.location.hash = ImportWords.Pages.ENTER_WORDS;
     }
 
     /**
-     * Store key and request passphrase
-     *
-     * @param {Array<string>} mnemonic
-     * @param {number | null} mnemonicType
+     * @param {string} [password = '']
+     * @returns {Promise<void>}
      */
-    _onRecoveryWordsComplete(mnemonic, mnemonicType) {
-        switch (mnemonicType) {
-        case Nimiq.MnemonicUtils.MnemonicType.BIP39: {
-            const entropy = Nimiq.MnemonicUtils.mnemonicToEntropy(mnemonic);
-            this._resolve(entropy);
-            break;
-        }
-        case Nimiq.MnemonicUtils.MnemonicType.LEGACY: {
-            const entropy = Nimiq.MnemonicUtils.legacyMnemonicToEntropy(mnemonic);
-            const privateKey = new Nimiq.PrivateKey(entropy.serialize());
-            this._resolve(privateKey);
-            break;
-        }
-        case Nimiq.MnemonicUtils.MnemonicType.UNKNOWN: {
-            this.entropy = Nimiq.MnemonicUtils.mnemonicToEntropy(mnemonic);
-            window.location.hash = ImportWords.Pages.CHOOSE_KEY_TYPE;
-            break;
+    async _storeKeys(password = '') {
+        TopLevelApi.setLoading(true);
+        let encryptionKey = null;
+        if (password && password.length > 7) {
```

Comment:
The length has a constant in the `PassphraseInput.DEFAULT_MIN_LENGTH = 8;`.

This check is maybe a bit unecessary, because the PassphraseSetterBox now has the password strength validation also and only fires its SUBMIT event when all conditions are met.